spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubTemplateIntegrationTests.java

```diff
@@ -157,9 +157,17 @@ public void testPullAndAck() {
 			});
 
 			Thread.sleep(11_000);
-			ackableMessages = pubSubTemplate.pull(subscriptionName, 4, true);
-			assertThat(ackableMessages.size()).as("check that we get both nacked messages back").isEqualTo(2);
-			ackableMessages.forEach(AcknowledgeablePubsubMessage::ack);
+			// pull the 2 nacked messages with retries for up to 10s
+			int messagesCount = 0;
+			int tries = 100;
+			while (messagesCount < 2 && tries > 0) {
```

Comment:
Do we still need to sleep those 11 seconds above with solution?

Reply:
We still do because we're trying to verify that the ack'ed message actually did get ack'ed and will not come back after 10 seconds. If it wasn't properly ack'ed, we will get 3 messages instead of 2.